birdhouse/env.local.example

```diff
@@ -142,12 +142,33 @@ export POSTGRES_MAGPIE_PASSWORD=postgres-qwerty
 # "Notebook" are all the tutorial notebooks on Jupyter.
 #export AUTODEPLOY_NOTEBOOK_FREQUENCY="@every 5m"
 
+# Log directory used for extra autodeploy tasks
+# export AUTO_DEPLOY_LOGDIR=/var/log/PAVICS
+
 # Add more jobs to ./components/scheduler/config.yml
 #
 # Potential usages: other deployment, backup jobs on the same machine
 #
 #export AUTODEPLOY_EXTRA_SCHEDULER_JOBS=""
 #
+# Example extra job that deploys custom notebooks for a specific image
+#export AUTODEPLOY_EXTRA_SCHEDULER_JOBS="
+#- name: notebookdeploy-eo
+#  comment: Auto-deploy tutorial notebooks for the eo image
+#  schedule: '${AUTODEPLOY_NOTEBOOK_FREQUENCY}'
+#  command: '/deploy-data-specific-image /notebook_config.yml.template'
```

Comment:
Following my previous comment regarding this config, given that this file `/notebook_config.yml.template` is in the image, why is it a .template?

Reply:
This "cronjob" is again pretty much similar to the existing https://github.com/bird-house/birdhouse-deploy/blob/57a320c3583f804838e8c18aeb064266527a0bc9/birdhouse/components/scheduler/deploy_data_job.env.  Same question, if a feature is missing, can we add to it than forking it?

This one you might not be aware since it's under the "scheduler" component because it has to be used with that component.

Using that "generic" cronjob allows a much simplified cronjob definition like this https://github.com/bird-house/birdhouse-deploy/blob/57a320c3583f804838e8c18aeb064266527a0bc9/birdhouse/components/scheduler/deploy_raven_testdata_to_thredds.env

Reply:
So, I used the template word because in this version of the config, we find the variable ${JUPYTERHUB_USER_DATA_DIR}, which gets replaced by its value when running the script. 
We do a copy of the config file using `envsubst < $TEMPLATE_CONFIG_YML > $CONFIG_YML`, where the copy has the real value instead of the variable.
I agree 'template' is not the best word here though. I don't know if you have a better idea.

In regards to your comment about the dest_dir variable found in the yaml config, I could remove it, and we could always output the notebooks to the same directory, in ${JUPYTERHUB_USER_DATA_DIR}/tutorial-notebooks/{$IMAGE_NAME}.
We could remove that option of customization, which could remove my need of having a copy of the config with variables to replace.
If we decide to remove the dest_dir option, I will just remove all mentions of .template, and remove the envsubst command from the script.

Reply:
Ok, it solves two issues and make the whole thing simpler, so I would remove the dest_dir option from the config.
But I would not use the directory ${JUPYTERHUB_USER_DATA_DIR}/tutorial-notebooks/{$IMAGE_NAME}. This is where the notebooks need to be on the host, but in the image it could be anywhere like /tmp as long as the calling command mount the host volume at that point. My suggestion still hold : in the docker command include the volume mount and set an environment variable telling the script where to put the stuff (the dest_dir, but as a env var rather than a config)

Reply:
Again, now that I have more time to digest this PR ...

I would suggest moving this sample big blob of code to generate the cronjob also into jupyter-pavics-base repo so it is together with the script `deploy-data-specific-image` it wraps.
* It makes the `env.local.example` shorter and less intimidating
* You now own the "code" so if you need to fix something (like add/remove a volume-mount) you can do it and it's transparent to all the callers, all they need is always run the latest version.  You basically provide the user with a stable interface and give you control to change the implementation.
* Later, in a different PR, once we "migrate" to the generic `deploy-data` script, the matching generic cronjob `components/scheduler/deploy_data_job.env` will also have to be adapted for whatever newer options that `deploy-data` will support.  And we still keep the same consistency that the "script" and the "cronjob wrapper" are together.

Try to inspire from the existing generic cronjob wrapper, especially the part about https://github.com/bird-house/birdhouse-deploy/blob/57a320c3583f804838e8c18aeb064266527a0bc9/birdhouse/components/scheduler/deploy_data_job.env#L64-L67 to ensure it also works during autodeploy (there is a slight difference between `./pavics-compose.sh` invoked manually on the console and invoke inside the scheduler component).

Reply:
Forgot to say, once you move the "cronjob generation code" out to jupyter-pavics-base, you can reference it back in `env.local.example` to "advertise" it, like https://github.com/bird-house/birdhouse-deploy/blob/57a320c3583f804838e8c18aeb064266527a0bc9/birdhouse/env.local.example#L171-L181